brian2/codegen/runtime/cython_rt/templates/spikegenerator.pyx

```diff
@@ -1,39 +1,45 @@
 {% extends 'common.pyx' %}
 
-{% block template_support_code %}
-cdef int _start_idx = 0
-cdef double _current_time = 0.0
-{% endblock %}
-
 {% block maincode %}
 
-    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time } #}
-     
-    global _start_idx, _current_time
-    
-    # Check if we have been reset, if so reset the cache
-    if t<_current_time:
-        _start_idx = 0
-    
-    _current_time = t
+    {# USES_VARIABLES {_spikespace, N, t, dt, neuron_index, spike_time, period, _lastindex } #}
 
     # TODO: We don't deal with more than one spike per neuron yet

```

Comment:
Well if we wanted to be fancy we could compare ``amax(timebins)`` against ``len(timebins)`` somehow and select one of the two algorithms. Maybe this is getting out of hand though? For your solution, for a 1h long set of spike times with dt=0.1ms (i.e. ``amax(timebins)==10000*60*60``), it would use 275MB of intermediate RAM. I doubt that this will ever be much of a problem, especially given that it gets deallocated immediately after. So I'd say let's go with your original bincount solution.